src/NuGet.Core/NuGet.Packaging/Rules/InvalidUndottedFrameworkRule.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Rules
+{
+    internal class InvalidUndottedFrameworkRule : IPackageRule
+    {
+        private const string LibDirectory = "lib";
```

Comment:
This is applicable in lib, ref, build and other folders as well. 

contentFiles also has a tfm part. 

If it's easier, you can do some of that in a future PR, but keep in mind there are other occurences.

Reply:
Yeah, I should be using `foreach (string knownFolder in PackagingConstants.Folders.Known)` here :)

Reply:
Just keep in mind that not *all* folders have the same structure. Ex. contentFiles.

Reply:
keep in mind the location of the TFM under `contentFiles` might be different to other paths.

I don't know if `tools\` is a known folder, but I don't think we look for TFM specific folders under there. I don't know if we should warn about a folder that looks like a net5 TFM there without a dot.